python/nano/src/bigdl/nano/pytorch/lightning.py

```diff
@@ -0,0 +1,66 @@
+#
+# Copyright 2016 The BigDL Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+from pytorch_lightning import LightningModule
+from torch import nn
+import torch
+from torch.nn.modules.loss import _Loss
+from torch.optim import Optimizer
+
+
+class LightningModuleFromTorch(LightningModule):
+    def __init__(self, model: nn.Module, loss: _Loss, optimizer: Optimizer):
+        """
+        Integrate pytorch modules, loss, optimizer to pytorch-lightning model.
+
+        :param model:       Pytorch model to be converted.
+        :param loss:        A torch loss function.
+        :param optimizer:   A torch optimizer.
+        """
+        super().__init__()
+        self._copy(model)
+        self.loss = loss
+        self.optimizer = optimizer
+
+    def _copy(self, torch_model: nn.Module):
+        for name, child in torch_model._modules.items():
+            setattr(self, name, child)
+        setattr(self, "forward", torch_model.forward)
+
+    def _forward(self, batch):
+        # Handle different numbers of input for various models
+        nargs = self.forward.__code__.co_argcount
+        return self.forward(*(batch[:nargs - 1]))
+
+    def training_step(self, batch, batch_idx):
+        x, y = batch
+        y_hat = self._forward(batch)
```

Comment:
I am sure I understand this method. 
Could we just call `self.forward` in `self._forward` and pass `x` to it?

Reply:
We have a few forward  with multiple inputs instead of one. Typically in Seq2Seq, during training the input can be (x, y) to use y as a prior, but input (x) during inference. Another case is that input is (x1,x2,x3,...,y). So in `self._forward` I'm trying to resolve this issue by getting the number of arguments. Another way is call `self.forward(batch)` and we have to handle some extra input like y when defining forward function, for example, using `forward(x, *args)`.

Reply:
for teacher forcing, can we let `Seq2Seq` override `train_step`. @TheaperDeng